Removing Singularity and related linting

matq007 opened the pull request:
Removing Singularity file together with linting functions related to it. I've also updated the doc. Should be everything hopefully. Implementation of issue #252.

matq007 commented:
I knew I should have tested before pushing 😿

ewels commented on docs/lint_errors.md:
```diff
@@ -209,50 +201,11 @@ These lines must be an exact copy of the above example.
-Additional lines and different metadata can be added without causing the test to fail.
-
-## Error #11 - Template TODO statement found ## {#11}
+## Error #10 - Template TODO statement found ## {#11}
```
Also you need to update the error ID in the line that throws the error / passes for this test. eg:

```python
self.failed.append((11, "template todo something was wrong or something: {}".format(missing)))
```

matq007 replied:
Isn't that defined here? https://github.com/nf-core/tools/blob/9f11213f2d60a6fdce86600ea086a4dc882daa8a/nf_core/lint.py#L766

ewels replied:
Yes - but it's number 11. You need to change it to number 10.

matq007 replied:
Oh, I see now ![Facepalm](https://upload.wikimedia.org/wikipedia/commons/thumb/3/3b/Paris_Tuileries_Garden_Facepalm_statue.jpg/300px-Paris_Tuileries_Garden_Facepalm_statue.jpg)

ewels commented on docs/lint_errors.md:
```diff
@@ -6,46 +6,44 @@ This page contains detailed descriptions of the tests done by the [nf-core/tools
 
 * `nextflow.config`
-  * The main nextflow config file
+    * The main nextflow config file
```
You reverted all of my markdown fixes! 😱 I think the markdown linter will fail with this as it is..

matq007 replied:
My editor ignored my settings 😱. Fixing

ewels commented:
This pull request **introduces 2 alerts** when merging 9f11213f2d60a6fdce86600ea086a4dc882daa8a into a1a7408ce7e7b4d502d29107bdea2cc406bb34f5 - [view on LGTM.com](https://lgtm.com/projects/g/nf-core/tools/rev/pr-bfd5aa87e43693576746033efc864c8010482015)

**new alerts:**

* 2 for Unused local variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

matq007 commented:
I had to adjust some values in the tests.
- I've removed 3 tests so that means `MAX_PASS_CHECKS` should be only `58`
- I've removed Singularity file meaning that from passed tests we have to subtract 5 and then 1

ewels commented:
Thanks! Yes the `MAX_PASS_CHECKS` thing is a bit of a hassle, currently just needs a manual update whenever you add or remove tests as you found. Not sure that there's a reliable better way to do it.

ewels commented on docs/lint_errors.md:
```diff
@@ -67,11 +65,11 @@ The following variables fail the test if missing:
   * The minimum version of Nextflow required to run the pipeline.
-  * Should `>=` a version number, eg. `manifest.nextflowVersion = '>=0.31.0'` (check the [Nexftlow documentation](https://www.nextflow.io/docs/latest/config.html#scope-manifest) for more.)
-  * This should correspond to the `NXF_VER` version tested by Travis.
+    * Should `>=` a version number, eg. `manifest.nextflowVersion = '>=0.31.0'` (check the [Nexftlow documentation](https://www.nextflow.io/docs/latest/config.html#scope-manifest) for more.)
```
Still lots of whitespace changes in this diff...?

ewels commented on nf_core/lint.py:
```diff
@@ -757,39 +737,10 @@ def check_conda_dockerfile(self):
         else:
             for missing in difference:
-                self.failed.append((10, "Could not find Singularity file string: {}".format(missing)))
+                self.failed.append((9, "Could not find Dockerfile file string: {}".format(missing)))
```
I'm pretty sure that the Dockerfile test is still number 10... right?

matq007 replied:
No, it's 9. https://github.com/matq007/tools/blob/master/docs/lint_errors.md

ewels replied:
Aha, nice! So this was an error in the previous code? Good spot 👍

ewels commented on docs/lint_errors.md:
```diff
@@ -83,9 +81,9 @@ The following variables throw warnings if missing:
-  * Should be set to a results folder, eg: `${params.outdir}/pipeline_info/trace.[workflowname].txt"``
-  * The DAG file path should end with `.svg`
-    * If Graphviz is not installed, Nextflow will generate a `.dot` file instead
+    * Should be set to a results folder, eg: `${params.outdir}/pipeline_info/trace.[workflowname].txt"``
```
```suggestion
  * Should be set to a results folder, eg: `${params.outdir}/pipeline_info/trace.[workflowname].txt"`
```

Still some whitespace left which breaks indentation. Also spotted this broken inline code block which was there previously.
